site/source/tutorials/1040_data_driven_widgets/index.md

```diff
@@ -0,0 +1,193 @@
+{% endaside %}
+
+The `ListProperties` interface now defines an optional `data` property that can be passed into a `List` widget. This property could've been called by any name other than `data`; the important takeaway is that data items can now be passed into the `List` widget using `properties`. The next step is to update the `List` to render a list of items based on its new `data` property.
+
+{% instruction 'Try to update the `List` so it also renders a `<div>` for each item in its `data` array property.' %}
+
+{% solution showsolution3 %}
+```ts
+protected renderItems() {
+    const { data = [] } = this.properties;
+    return data.map((item: any) => v('div', [ `${item.firstName} ${item.lastName}` ]));
+}
+
+protected render() {
+    return v('div', [
+        w(TextInput, {
+            value: this.properties.value,
+            onInput: this.onInput,
+            placeholder: 'Filter workers...'
+        }),
+        v('div', this.renderItems())
+    ]);
+}
+```
+{% endsolution %}
+
+The `List` widget now renders both a `TextInput` that a user can type into and a list of result items based on its `data` property. Item rendering is offloaded into a helper method (`renderItems`) so widgets that extend `List` that need to modify how items are rendered only have to override a small helper method instead of the main `render` method.
+
+The next step to update the `Banner` widget to pass the correct `data` into the `List`.
+
+{% task 'Pass worker data down through the `Banner` to the `List`' %}
+
+Currently, the application keeps all worker data as a private variable within the `App` widget. In order to use this data to also power the `List`, the `Banner` widget must also be updated to accept a `data` property so it can in turn pass it down to the `List` it renders.
+
+{% instruction 'Export a `BannerProperties` interface from the `Banner` widget so a `data` array property can be passed in:' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/Banner.ts' lines:4,5-8 %}
+
+{% instruction 'Update the generic parameter passed into `WidgetBase` to use the new interface:' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/Banner.ts' lines:10 %}
+
+{% instruction 'Pass worker data to the routing outlet for the `Banner` widget in `App.ts`:' %}
+
+```ts
+w(BannerOutlet, {
+    data: this._workerData
+})
+```
+
+Now that the `Banner` widget has worker data, it can pass a subset of this data into the `List` widget based on the current `List` input value.
+
+{% instruction 'Pass a subset of worker data into the `List` based on its `value` in `Banner.ts`:' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/Banner.ts' lines:11-35 %}
+
+A helper method was added that filters worker data items by name based on a query value. These filtered items are passed to the `List` as its `data`, causing it to render result items if any exist based on user input.
```

Comment:
causing it to render result items if any exist based on user input -> causing the `List` to render result items if any items exist based on the user specified filter